src/extension-runners/firefox-android.js

```diff
@@ -131,6 +133,10 @@ export class FirefoxAndroidExtensionRunner {
     await this.adbDevicesDiscoveryAndSelect();
     await this.apkPackagesDiscoveryAndSelect();
     await this.adbCheckRuntimePermissions();
+    if (adbCleanArtifacts) {
+      await this.adbArtifactsDir(true);
+    }
+    await this.adbArtifactsDir(false);
```

Comment:
In my opinion we should only do this once, we clean all the old artifacts dirs or we just check if there are any.